### PR TITLE
Unhardcode a bunch of shipyard console stuff

### PR DESCRIPTION
**Pull request by arimah**
## About the PR
Move a bunch of stuff from ShipyardSystem to the ShipyardConsoleComponent, to reduce hardcoded nonsense. Chiefly this reduces logic based on the UI key (very smelly code).

* `ShipyardChannel` is now *always* used for the primary purchase and sale announcement, instead of picking the channel based on the UI key.
* `SecurityShipyardChannel` is gone, and in its place we get `SecretShipyardChannel`, a secondary channel that announces redacted information. Used by black market and syndicate shipyards to alert the NFSD.
* New field: `NewJobTitle`, which updates the ID card's job title if present. Allows us to *not* update job titles for NFSD and the SR. Additional logic will be introduced **in a future PR** such that we *always* leave certain jobs alone.
* New field: `NewAccessLevels`. Pretty self-explanatory. Reduces hardcoding.
* New field: `SalesTax`, for black market and syndicate ships. If set to anything but 0, the owner is taxed this percentage of the shuttle's value when the ship is sold, and the tax levied is credited to the station, as before.
* Change strings to `ProtoId` where appropriate so we get validation and nice serialization for free.
* There is a new abstract prototype `ComputerShipyardBase` that defines everything needed for a basic shipyard console, and `ComputerShipyard` extends it with what's necessary for civilian ships.

## Why / Balance
The shipyard console code was a downright mess of hardcoded stuff in places.

## How to test
1. Spawn in every shipyard console.
2. Equip yourself with a CentCom headset, then give it freelance and syndicate comms.
3. Insert a fresh passenger ID card into each shipyard console, buy a ship, sell it.
4. For the NFSD, security and Empress shipyard consoles, verify that:
    * The job title remains "Passenger"; and
    * The ID card has Captain, Brig and Security access; and
    * The purchase and sale are announced in the NFSD channel only.
5. For the black market and syndicate shipyard consoles, verify that:
    * The job title is changed to "Captain"; and
    * The ID card has Captain access; and
    * The purchase and sale are announced in the Freelance and Syndicate channels, respectively, and in the NFSD channel with redacted information; and
    * The sales tax is applied correctly – the shuttle should be nearly worthless, and the station should be credited with money upon selling the shuttle.
6. For every other shipyard console, verify that:
    * The job title is changed to "Captain"; and
    * The ID card has Captain access; and
    * The purchase and sale are announced in the Traffic channel only.
7. Verify that the black market and syndicate shipyard consoles can be destroyed.
8. Verify that no shipyard console can be unanchored or deconstructed (should block all tool use).

I have done all these things myself and everything *seems* to be working fine.

## Media
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
The Empress shipyard console now no longer sets your job title to "Captain". This is technically a breaking change, but should not have any real effect.

**Changelog**
:cl:
- fix: Empress shipyard console no longer sets users' job ID to Captain.

### CODE REVIEW

**Comment by dvir001** [comment]
Watch this break upstream merge FR FR

Still need to merge it before the upstream merge.

File: Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs
Comment (whatston3) [inline Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs:322]:
```suggestion
        if (component.SalesTax > 0.0f)
```
Equality check doesn't play nicely with NaN.  If we need bonuses on selling ships, this needs revision.

File: Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs
Comment (whatston3) [inline Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs:377]:
```suggestion
        if (component.SalesTax > 0.0f)
```
Equality check doesn't play nicely with NaN.  If we need bonuses on selling ships, this needs revision.

File: Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs
Comment (whatston3) [inline Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs:465]:
```suggestion
            if (component.SalesTax > 0.0f)
```
Equality check doesn't play nicely with NaN.  If we need bonuses on selling ships, this needs revision.

File: Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs
Comment (whatston3) [inline Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs:233]:
```suggestion
        if (component.SalesTax > 0.0f)
```
Equality check doesn't play nicely with NaN.  If we need bonuses on selling ships, this needs revision.

File: Resources/Prototypes/_NF/Entities/Structures/Machines/Computers/computers_shipyard_mothership.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Structures/Machines/Computers/computers_shipyard_mothership.yml:21]: Regarding the comment above, wouldn't multiple inheritance fix this?

Need to double check w.r.t. inheritance ordering (assuming order runs left-to-right)
```suggestion
  parent: [BaseMothershipComputer, ComputerShipyardNfsd]
```

Might want to add a separate suffix and description here that's empress-specific.

File: Resources/Prototypes/_NF/Entities/Structures/Machines/Computers/computers_shipyard.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Structures/Machines/Computers/computers_shipyard.yml:139]: Might be dumb, but is this defining new behaviour or inheriting from what already exists?  Seems sane, only issue I could imagine is if a prisoner or other antagonist was able to buy a ship, not sure if that's possible OTOH.

**Review by whatston3 — CHANGES_REQUESTED** [review]
A few minor remarks.  Looks good, need to run some local tests.

File: Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs
Comment (whatston3) [reply Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs:322]: Actually, since this same check gets repeated a few times, it'd be nice to put into a function, `HasValidSalesTax()` or the like.

File: Resources/Prototypes/_NF/Entities/Structures/Machines/Computers/computers_shipyard.yml
Comment (arimah) [reply Resources/Prototypes/_NF/Entities/Structures/Machines/Computers/computers_shipyard.yml:139]: Inheriting existing behaviour:

https://github.com/new-frontiers-14/frontier-station-14/blob/e77ade7db7d8d0a9c68f865457d773990689eb9f/Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs#L178-L182

File: Resources/Prototypes/_NF/Entities/Structures/Machines/Computers/computers_shipyard_mothership.yml
Comment (arimah) [reply Resources/Prototypes/_NF/Entities/Structures/Machines/Computers/computers_shipyard_mothership.yml:21]: That does not seem to work correctly, unfortunately. But you did highlight an error in my implementation: description and Mothership suffix are not present with the current version. I'll see if I can fix this in... some way, without introducing a bajillion new prototypes.

File: Content.Shared/Shipyard/Components/SharedShipyardConsoleComponent.cs
Comment (arimah) [reply Content.Shared/Shipyard/Components/SharedShipyardConsoleComponent.cs]: It shouldn't be! That's a development artifact that managed to sneak in, how very sneaky. Fixed now.

File: Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs
Comment (arimah) [reply Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs]: I moved sales tax into a new method called `CalculateSalesTax`, which uses `float.IsFinite` to exclude NaN and infinities.

File: Resources/Prototypes/_NF/Entities/Structures/Machines/Computers/computers_shipyard_mothership.yml
Comment (arimah) [reply Resources/Prototypes/_NF/Entities/Structures/Machines/Computers/computers_shipyard_mothership.yml]: I pushed an update that extends `BaseMothershipComputer` only, and duplicates a little bit of stuff from the NFSD shipyard console. It was the simplest and most reliable way to get functionality from both, without introducing a load of annoying extra stuff.

**Review by whatston3 — COMMENTED** [review]
Looks good, running a quick test, will approve & merge if everything seems fine.

**Review by whatston3 — APPROVED** [review]
Looks good, works sensibly, job changes as expected.

Unless it's my mouse or something weird with my test setup, I wasn't aware that the security shipyard console spams you with four or so popups on exiting the UI.  Noting that down, but it's unrelated to this.

File: Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs
Comment (arimah) [inline Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs:42]: What the hell, where did this come from? Ugh. I can make a new PR to remove this stupid extra import.

**Pull request merged**
